src/adapter/src/coord/sequencer.rs

```diff
@@ -1103,18 +1103,6 @@ impl<S: Append + 'static> Coordinator<S> {
                     .await
                     .unwrap();
 
-                // We must advance the timeline to `since_ts` so that the table is not invalid.
```

Comment:
The idea here is that instead of advancing the whole timeline to `since_ts` and potentially blocking queries on objects whose uppers is less than `since_ts`, we:
1. Don't advance the timeline.
2. Get a read hold on `table_id` at `since_ts` (this will happen in `initialize_storage_read_policies`).
3. Any query on `table_id` will block in Strict Serializable mode until the timeline has advanced to `since_ts`.

By the way, I'm actually convinced that this was a bug. We were updating the read holds to a new time, but we weren't updating the timestamp oracle to a new time. That means there was a small window where we could query at the timestamp oracle's read timestamp, and that timestamp would not be valid because the object had been compacted up to `since_ts`. We should have also been advancing the timeline. It doesn't really matter now though because we are removing it.